tp/src/state.rs

```diff
@@ -171,19 +176,19 @@ impl<'a> SabreState<'a> {
         let d = self.context.get_state_entry(&address)?;
         match d {
             Some(packed) => {
-                let contract_registries: ContractRegistryList =
-                    match protobuf::parse_from_bytes(packed.as_slice()) {
-                        Ok(contract_registries) => contract_registries,
-                        Err(err) => {
-                            return Err(ApplyError::InternalError(format!(
-                                "Cannot deserialize contract registry list: {:?}",
-                                err,
-                            )));
-                        }
-                    };
+                let contract_registries = match ContractRegistryList::from_bytes(packed.as_slice())
+                {
+                    Ok(contract_registries) => contract_registries,
+                    Err(err) => {
```

Comment:
This sort of statement would be better with a `.map_err(|e| ...)?` style, removing the `match` and `Ok` branches.